src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -1128,7 +1128,10 @@ public override ClaimsPrincipal ValidateToken(string token, TokenValidationParam
             ValidateConditions(samlToken, validationParameters);
             var issuer = ValidateIssuer(samlToken.Issuer, samlToken, validationParameters);
             ValidateTokenReplay(samlToken.Assertion.Conditions.NotOnOrAfter, token, validationParameters);
-            ValidateIssuerSecurityKey(samlToken.SigningKey, samlToken, validationParameters);
+            if (validationParameters.RequireSignedTokens || samlToken.SigningKey != null)
```

Comment:
Can we put this logic into Validators.ValidateIssuerSecurityKey?

Reply:
I was following the original issue description: `Our handlers should not call ValidateIssuerSigningKey...`
ValidateIssuerSecurityKey is a public static method so I assumed that it's safer only to change our handlers.

I can centralize this change by putting the logic in ValidateIssuerSecurityKey, if we are fine with altering the ValidateIssuerSecurityKey method.

Reply:
That title wasn't accurate.
The issue is we are throwing when we shouldn't.
The pattern we locked on is: SecurityTokenHandler-> Virtual -> Validators.
In this way we will have the same results across token types.
It is this line here that is causing us grief: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/eac57a0465f9d76d1c6a840c207e20084cb7de07/src/Microsoft.IdentityModel.Tokens/Validators.cs#L203

The fix looks good. Should be in validators.
Saml2SecurityTokenHandler is fine, but SamlSecurityTokenHandler should just call into the Validator.

Reply:
addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1158/commits/9e94ffea506bcab20e4cede543525783e149715c.